### PR TITLE
apiv2: set mitigated date if applicable

### PR DESCRIPTION
**Pull request by keenan-v1**
- Sets or clears mitigated date based on the value of is_Mitigated
- Sets or clears mitigated_by based on the value of is_Mitigated and current user context

### CODE REVIEW

**Review by valentijnscholten — CHANGES_REQUESTED** [review]
thanks. i was confused a bit, but the logic looks ok. could you make it so we have 2 booleans and call save if one of them is True. so 'push_to_jira' and 'set_mitigated'? now if you quickyl read it could make readers think in some cases we will not save the instance at all due to the naming of the boolean.

**Comment by valentijnscholten** [comment]
should the same logic be in perform_create? if somebody creates a finding that is already marked as mitigated?

File: dojo/api_v2/serializers.py [hunk 1]
```diff
@@ -693,9 +693,25 @@ def update(self, instance, validated_data):
...
+        save_instance = push_to_jira
+
+        # Allow setting or clearing the mitigation date based upon the state of is_Mitigated.
+        if instance.is_Mitigated and instance.mitigated is None:
+            save_instance = True
+            instance.mitigated = datetime.datetime.now()
+            instance.mitigated_by = self.context['request'].user
+            if settings.USE_TZ:
+                instance.mitigated = timezone.make_aware(instance.mitigated, timezone.get_default_timezone())
+        elif not instance.is_Mitigated and instance.mitigation is not None:
```
Comment (valentijnscholten) [inline dojo/api_v2/serializers.py:706]: should this check for `instance.mitigated` instead of `instance.mitigation` ?

Comment (keenan-v1) [reply dojo/api_v2/serializers.py:706]: Good catch. My testing didn't catch this. Will fix.

**Comment by keenan-v1** [comment]
I've added the requested changes and fixed the bug in variable naming. Tested the functionality and it is working as intended.

**Review by valentijnscholten — APPROVED** [review]

**Comment by valentijnscholten** [comment]
thanks for the PR

**Comment by madchap** [comment]
As a push to jira will actually overwrite the jira description field, this update may come as a surprise to some users. On the UI, there is the `?` that warns about it, but in this case it will just do it.

**Comment by keenan-v1** [comment]
> 
> 
> As a push to jira will actually overwrite the jira description field, this update may come as a surprise to some users. On the UI, there is the `?` that warns about it, but in this case it will just do it.

Isn't that more a problem with the push-to-jira patch and less of a problem with this one? I'm not entirely sure how the Jira stuff works as I don't use it yet. I simply worked around the Jira code here as it was overriding the same areas as my original fix for the mitigated date did.

**Review by madchap — APPROVED** [review]

**Pull request merged**